### DISCUSSION TITLE
How to Add Custom Categories Without Losing Default Categories or Overloading Labels?

### DISCUSSION THREAD

**Discussion by tanhaow** [mem0ai/mem0#2080] [Q&A]
I’m using Mem0’s custom categories, but I’m having trouble keeping the default categories and adding new ones without all categories showing up on every memory. Here’s what I’ve noticed:

- If I only define custom categories, I lose the default ones.
- If I include both default and custom categories together, every memory ends up tagged with all categories, even if they’re irrelevant (see pic below).

<img width="993" alt="image" src="https://github.com/user-attachments/assets/25bbfeee-6aea-4d6b-b4b6-6afef6ff9d63">

**Questions:**
1. How can I add a few custom categories without having to redefine all the default categories?
2. How do I prevent Mem0 from tagging every memory with every category, showing only the relevant ones?

Any guidance would be appreciated! 🙏🙏🙏

**My code:**
<pre> ```

def add_memory(phone_number: str, role: str, content: str):
    messages = [{"role": role, "content": content}]
    
    custom_categories = [
        # Default categories redefined here
        {"personal_details": "User's identity or personal info."},
        {"family": "Details about user's family."},
        {"professional_details": "User's professional life."},
        {"sports": "User's interest in sports."},
        {"travel": "User's travel-related info."},
        {"food": "Dietary preferences, cooking, favorite cuisines."},
        {"music": "User's musical tastes and experiences."},
        {"health": "User's health, wellness, medical info."},
        {"technology": "User's interests and experiences with tech."},
        {"hobbies": "User's leisure activities and hobbies."},
        {"fashion": "User's clothing/style preferences."},
        {"entertainment": "User's interests in TV, movies, etc."},
        {"milestones": "Key life events or personal milestones."},
        {"user_preferences": "General likes/dislikes not covered above."},
        {"misc": "Uncategorized or miscellaneous info."},
        
        # My custom category
        {"call_schedule": "Mentions of best call times, call preferences, etc."}
    ]

    try:
        mem0_client.add(
            messages=messages,
            user_id=phone_number,
            custom_categories=custom_categories,
        )
    except Exception as e:
        print(f"Error adding memory: {e}")

``` </pre>

**Comment by Dev-Khant** [comment]
Hey @rachel-tanhao, we've improved this feature and will be releasing it soon. With this update, if you set custom categories at the project level, all memories added to that project will automatically be assigned to the relevant custom categories, ensuring better relevance for the memories.
